packages/apollo/src/commands/client/extract.ts

```diff
@@ -40,52 +24,31 @@ export default class ClientExtract extends ClientCommand {
   ];
 
   async run() {
-    const { clientIdentity, operations, filename }: any = await this.runTasks(
-      ({ flags, project, config, args }) => [
-        {
-          title: "Extracting operations from project",
-          task: async ctx => {
-            const operations = Object.values(
-              this.project.mergedOperationsAndFragmentsForService
-            ).map(operationAST => {
-              // While this could include dropping unused definitions, they are
-              // kept because the registered operations should mirror those in the
-              // client bundle minus any PII which lives within string literals.
-              const printed = printWithReducedWhitespace(
-                sortAST(hideCertainLiterals(operationAST))
-              );
-
-              return {
-                signature: manifestOperationHash(printed),
-                document: printed,
-                metadata: {
-                  engineSignature: engineSignature(operationAST)
-                }
-              };
-            });
-
-            ctx.operations = operations;
-            ctx.clientIdentity = config.client;
-          }
-        },
-        {
-          title: "Outputing extracted queries",
-          task: (ctx, task) => {
-            const filename = args.output;
-            task.title = "Outputing extracted queries to " + filename;
-            ctx.filename = filename;
-            writeFileSync(
-              filename,
-              JSON.stringify(
-                { version: 1, operations: ctx.operations },
-                null,
-                2
-              )
-            );
-          }
+    const { clientIdentity, operations, filename } = await this.runTasks<{
+      clientIdentity: ClientIdentity;
+      operations: ManifestEntry[];
+      filename: string;
+    }>(({ flags, project, config, args }) => [
+      {
+        title: "Extracting operations from project",
+        task: async ctx => {
+          ctx.operations = getOperationManifestFromProject(this.project);
+          ctx.clientIdentity = config.client;
         }
-      ]
-    );
+      },
+      {
+        title: "Outputing extracted queries",
+        task: (ctx, task) => {
+          const filename = args.output;
+          task.title = "Outputing extracted queries to " + filename;
+          ctx.filename = filename;
+          writeFileSync(
+            filename,
+            JSON.stringify({ version: 1, operations: ctx.operations }, null, 2)
```

Comment:
It's possible we should bump this `version` to `2`, even with the current changes.

Reason being: currently, the `apollo-server-plugin-operation-registry` will `throw` if the manifest it tries consuming is `version !== 1`, so this should allow us to ensure that the new `apollo-server-plugin-operation-registry` which starts consuming this will use the same pre-hash normalization algorithm against incoming operations.

If we do bump this, I definitely think we should change the `signature` attribute on the individual documents in the `operations[]` to be `sha256Hash` as @glasser has proposed.

Reply:
Going to go forward with the version bump, but for now, changing the shape of `operations[]` _does_ require a change to the backend (see: `RegisteredOperationInput` type).

My intuition says to defer this until 2.0 work in order to keep the scope of this work within reason, but I'm open to being convinced otherwise if you believe this is important enough.

Reply:
Resolved in f7915e1d5199f8e89957f4cd0b507659b5ff8d02

Reply:
Deferring the change of `signature` seems fine.  We also need to update the backend to support a different `version` other than `1` (not in open-source land!).

Reply:
Added to overall TODOs and will leave this issue unresolved until I address it on the backend 👍